applications/dashboard/models/class.logmodel.php

```diff
@@ -452,14 +452,38 @@ public function Recalculate() {
          return;
 
       $In = implode(',', array_keys($DiscussionIDs));
-      if (empty($In))
-         return;
-      
-      $Px = Gdn::Database()->DatabasePrefix;
-      $Sql = "update {$Px}Discussion d set d.CountComments = (select coalesce(count(c.CommentID), 0) + 1 from {$Px}Comment c where c.DiscussionID = d.DiscussionID) where d.DiscussionID in ($In)";
-      Gdn::Database()->Query($Sql);
+      if (!empty($In)) {
+
+         $Px = Gdn::Database()->DatabasePrefix;
+         $Sql = "update {$Px}Discussion d set d.CountComments = (select coalesce(count(c.CommentID), 0) + 1 from {$Px}Comment c where c.DiscussionID = d.DiscussionID) where d.DiscussionID in ($In)";
+         Gdn::Database()->Query($Sql);
+         $this->_RecalcIDs['Discussion'] = array();
+      }
+
+      $UserIDsComment = $this->_RecalcIDs['UserComment'];
+
+      $In = implode(',', array_keys($UserIDsComment));
+      if (!empty($In)) {
+
+         $Px = Gdn::Database()->DatabasePrefix;
+         $Sql = "update {$Px}User set CountComments = coalesce(CountComments, 0) + 1 where UserID in ($In)";
```

Comment:
While I understand the original sub-select using `Query()` directly (it's not immediately obvious to me if it's even possible otherwise in our SQL driver), I'm not sure these more straight-forward statements warrant it. Naked SQL is technical debt.